project/backend/src/main/java/com/google/sps/servlets/GenerateSwipeMatchGroupIdServlet.java

```diff
@@ -0,0 +1,89 @@
+package com.google.sps.servlets;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/swipe-match/generate-group-id")
+public class GenerateSwipeMatchGroupIdServlet extends HttpServlet {
+
+  private static final String PROJECT_ID = "mak-step-2020";
+  private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  private static final String NUMBERS = "0123456789";
+  private static final int GROUP_ID_LENGTH = 8;
+  private static final String COLLECTION_NAME = "swipe-match-sessions";
+  private static final int RETRY_LIMIT = 10;
+  
+  private FirestoreOptions firestoreOptions;
+  private Firestore db;
+
+  private class RetryLimitExceededException extends Exception {
+    public RetryLimitExceededException(String message) {
+      super(message);
+    }
+  }
+
+  @Override
+  public void init() {
+    try {
+      firestoreOptions = FirestoreOptions.getDefaultInstance().toBuilder()
+        .setProjectId(PROJECT_ID)
+        .setCredentials(GoogleCredentials.getApplicationDefault())
+        .build();
+      
+      db = firestoreOptions.getService();
+    } catch (IOException e) {
+      // Not sure what to do if initialization fails...
+    }
+  }
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws IOException {
+    try {
+      String groupId = generateGroupId(0);
+      response.getWriter().println(groupId);
+    } catch (RetryLimitExceededException e) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  private String generateGroupId(int count) throws RetryLimitExceededException {
+    if (count == RETRY_LIMIT) {
+      throw new RetryLimitExceededException("Group ID generation retry limit exceeded.");
+    }
+
+    String groupId = "";
+    for (int i = 0; i < GROUP_ID_LENGTH / 2; i++) {
+      groupId = ALPHABET.charAt((int) Math.floor(Math.random() * ALPHABET.length())) + groupId;
+      groupId += NUMBERS.charAt((int) Math.floor(Math.random() * NUMBERS.length()));
+    }
+
+    // TODO: remove
+    groupId = "ABCD1234";
+
+    if (isGroupIdUnique(groupId)) {
+      return groupId;
+    } else {
+      return generateGroupId(count + 1);
+    }
+  }
+
+  private boolean isGroupIdUnique(String groupId) {
+    try {
+      return !db.collection(COLLECTION_NAME).document(groupId).get().get().exists();
+    } catch (InterruptedException e) {
+      return false;
+    } catch (ExecutionException e) {
+      return false;
+    }
```

Comment:
nit: we can merge these catch blocks into one as they're handled in the same way 
catch (InterruptedException | ExecutionException e)